misk/src/main/kotlin/misk/web/Request.kt

```diff
@@ -3,13 +3,53 @@ package misk.web
 import misk.web.actions.WebSocket
 import okhttp3.Headers
 import okhttp3.HttpUrl
+import okhttp3.MediaType
+import okhttp3.OkHttpClient
+import okhttp3.Protocol
+import okhttp3.Response
+import okhttp3.ResponseBody
+import okio.BufferedSink
 import okio.BufferedSource
 import org.eclipse.jetty.http.HttpMethod
+import java.io.IOException
+import java.net.HttpURLConnection
 
 data class Request(
   val url: HttpUrl,
-  val method: HttpMethod,
+  val method: HttpMethod = HttpMethod.GET,
   val headers: Headers = Headers.of(),
   val body: BufferedSource,
   val websocket: WebSocket? = null
 )
+
+fun Request.toOkHttp3(newUrl: HttpUrl?): Response {
+  val client = OkHttpClient()
+
+  val okUrl = if (newUrl != null) newUrl else this.url
+  val okRequestBody = if (this.method == HttpMethod.GET) null else object : okhttp3.RequestBody() {
+    override fun contentType(): MediaType? = null
+    override fun writeTo(sink: BufferedSink) {
+      sink.writeAll(body)
+    }
+  }
+
+  val okRequest = okhttp3.Request.Builder()
+      .url(okUrl)
+      .method(this.method.asString(), okRequestBody)
+      .headers(this.headers)
+      .build()
+
+  try {
+    return client.newCall(okRequest).execute()
+  } catch (e: IOException) {
+    return Response.Builder()
```

Comment:
would life be simpler if we never create an okhttp3.Response in this case? Ie. if we went directly to the Misk Response? Right now we’re creating an okhttp3.Response but we’re always going to immediately convert it to a Misk one anyway

Reply:
tradeoff between having single function do all the work vs more generic function that others could use. Though, is there a foreseeable use case where someone would convert a Misk request -> okHttp but then *not* return a Misk response from the okHttp response?

Reply:
Refactor now returns Misk response directly.